helix-lock/src/main/java/org/apache/helix/lock/HelixLock.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.helix.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Generic interface for Helix distributed lock
+ */
+public interface HelixLock {
+  /**
+   * Acquire a lock
+   * @return true if the lock was acquired, false if could not be acquired
+   */
+  public boolean acquireLock();
```

Comment:
- Do we want `public` modifier here because the interface is already `public`? I don't think it is necessary. All abstract, default, and static methods in an interface are implicitly public, so you can omit the public modifier.
- Do we really need to return a boolean? I think it makes sense to return a boolean for a `tryAcquire`, but for `acquire`, I think void is enough and clean.
- How about just naming `acquire`
- Do we want a `tryAcquire` with timeout?

Reply:
- Good catch, will remove `public`.

- This interface is a generic interface for both blocking and nonblocking blocks. In the design, if a false is return for `acquireLock`, it means the user failed to acquire the lock; a returned true means the lock is successfully acquired. A user cannot work on the resource without holding the lock.

- I personally think `acquireLock` is more clear that the user is using a lock, even if the variable name does not show it's a lock

- For blocking lock, we will not expose the `tryAcquire` to our users, `tryAcquire` and `retry` will be wrapped inside the `acquireLock` implementation.